### DISCUSSION TITLE
Testing RelationManager Form Fill

### DISCUSSION THREAD

**Discussion by AndyTWF** [filamentphp/filament#3434] [Help] [answered]
I have the following code in a testcase:

```
        Livewire::test(NavaidResource\RelationManagers\HoldsRelationManager::class, ['ownerRecord' => $navaid])
            ->callTableAction(ViewAction::class, $hold)
            ->assertSet('description', $hold->description);
```

This is on a HasMany relation where the RelationManager has View, Edit and Delete actions on the table.

I can see from debugging that the record -> form mutations are being run as expected, however whenever I run the assert - I can't find the appropriate key to be looking in to test that the form is being populated (though I can see data in the `dd` when dumping the instance, just can't seem to get at it for the test).

Is it possible to test for form filling on relation managers, or have I just been looking in the wrong place? :D

Thanks :)

**Comment by danharrin** [comment, marked as answer]
Just opened #3437, I'm adding some more testing helpers for you to accomplish this

**Reply by AndyTWF** [reply]
Awesome! Thanks :)

**Reply by danharrin** [reply]
Hey! I've finished the new methods, added docs, and merged the PR. They will be in the next release.

**Reply by AndyTWF** [reply]
Thanks! Have tested it and it works a treat :)
